pkg/pgmodel/model/batch.go

```diff
@@ -68,3 +68,15 @@ func (t *Batch) Visitor() *batchVisitor {
 func (t *Batch) Absorb(other Batch) {
 	t.AppendSlice(other.data)
 }
+
+func (t *Batch) Len() int {
+	return t.CountSeries()
+}
+
+func (t *Batch) Swap(i, j int) {
+	t.data[i], t.data[j] = t.data[j], t.data[i]
+}
+
+func (t *Batch) Less(i, j int) bool {
+	return t.data[i].Series().seriesID < t.data[j].Series().seriesID
```

Comment:
We should make sure the seriesID is set. GetSeriesID() is safer
```suggestion
	s1, _, err := t.data[i].Series().GetSeriesID() 
	if err != nil {
	  panic()
	}
	s2 _, err :=  t.data[j].Series().GetSeriesID()
	if err != nil {
	  panic()
	}
	return s1 < s2
```